p2p/discovery/node.go

```diff
@@ -52,7 +71,22 @@ func NewNode(id NodeID, ip net.IP, port uint16) *Node {
 		ID:      id,
 		IP:      ip,
 		UDPPort: port,
+	}
+}
 
-		sha: id.ToSha(),
+func getRandomNodeID() NodeID {
+	keypair, err := crypto.GenerateKey()
+	if err != nil {
+		log.Info(err.Error())
 	}
+
+	buff := crypto.FromECDSAPub(&keypair.PublicKey)
+
+	id, err := BytesToID(buff[1:])
```

Comment:
why exclude the first byte in buff?

Reply:
the first byte is a magic code, maybe we could refine the crypto method later